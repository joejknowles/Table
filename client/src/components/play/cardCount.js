import React from 'react';
import { connect } from 'react-redux';

import res from '../../resources/pages/play';

import { cardCountSelector } from '../../reducers';

export const CardCount = ({ count }) => (
  <div className="card-count">
    { res.cardCount(count) }
  </div>
);

const mapStateToProps = (state) => ({
  count: cardCountSelector(state)
});

export default connect(mapStateToProps)(CardCount);
