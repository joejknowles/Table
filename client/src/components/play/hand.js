import React from 'react';

import PlayCardButton from './playCardButton';

export default () => (
  <div className='card'>
    <PlayCardButton />
  </div>
);
