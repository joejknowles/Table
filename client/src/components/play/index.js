import React from 'react';
import { connect } from 'react-redux';

import Hand from './hand';
import { handSelector } from '../../reducers';
import '../../styles/Play.css';
import res from '../../resources/pages/play';

export const Play = ({ hand }) => (
  <div className="App Play">
    <div className="deck">
      { hand > 0 ?
          <Hand /> :
          <p className='no-cards-message'>
            { res.noCardsMessage }
          </p>
      }
    </div>
  </div>
);


export default connect(
  (state) => ({ hand: handSelector(state) })
)(Play);
