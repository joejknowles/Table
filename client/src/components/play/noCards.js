import React from 'react';
import res from '../../resources/pages/play';

export default () => (
  <div className='pile'>
    <div className="noCards">
        <p className='no-cards-message'>
          { res.noCardsMessage }
        </p>
    </div>
  </div>
);
