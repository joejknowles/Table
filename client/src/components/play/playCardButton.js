import React from 'react';
import { DispatchButton } from '../common/buttons';
import res from '../../resources/pages/play';

export default ({ dispatch }) => (
  <DispatchButton
    res={ res.playCard }
    className='play-card btn'
    action={ { type: 'PLAY_CARD' } }
  />
);
