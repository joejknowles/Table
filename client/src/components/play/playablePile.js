import React from 'react';
import PlayCardButton from './playCardButton';
import SwipeablePile from './swipeablePile';

function is_touch_device() {
  if ( 'callPhantom' in window ) return false; // integration tests fail without this
  return 'ontouchstart' in window ||        // works on most browsers
       navigator.maxTouchPoints;       // works on IE10/11 and Surface
};

export default () => (
    is_touch_device() ?
      <SwipeablePile />
        :
      <PlayCardButton />
);
