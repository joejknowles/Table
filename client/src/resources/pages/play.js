export default {
  playCard: 'Play Card',
  noCardsMessage: 'No more cards',
  youWin: 'You win!',
  snap: 'Snap!',
  cardCount: (count) => `${ count } cards remaining`
};
