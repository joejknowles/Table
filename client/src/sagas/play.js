import { call, takeEvery } from 'redux-saga/effects';

import * as toPath from '../routing';
import * as events from '../api/sockets';

export function* playCard(socket, action) {
  yield call(events.playCard, socket, action.card)
}

export function* playerBegin(socket) {
  yield takeEvery('PLAY_CARD', playCard, socket);
  yield call(toPath.play);
}

export function* playerJoin(socket) {
  yield call(events.joinPlayersRoom, socket);
  yield takeEvery('BEGIN_GAME', playerBegin, socket);
  yield call(toPath.waiting);
}
