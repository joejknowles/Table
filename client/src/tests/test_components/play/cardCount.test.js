import { CardCount } from '../../../components/play/cardCount';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('CardCount renders as before', () => {
  const component = shallowRenderer.render(<CardCount count={ 5 } />);
  expect(component).toMatchSnapshot();
});


it('CardCount renders as before', () => {
  const component = shallowRenderer.render(<CardCount count={ 0 } />);
  expect(component).toMatchSnapshot();
});
