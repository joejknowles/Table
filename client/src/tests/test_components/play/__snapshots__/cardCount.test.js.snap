exports[`test CardCount renders as before 1`] = `
<div
  className="card-count">
  5 cards remaining
</div>
`;

exports[`test CardCount renders as before 2`] = `
<div
  className="card-count">
  0 cards remaining
</div>
`;
